spring/boot3-autoconfigure/src/test/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationUtilTest.java

```diff
@@ -0,0 +1,56 @@
+package com.linecorp.armeria.internal.spring;
+
+import static com.linecorp.armeria.internal.spring.ArmeriaConfigurationUtil.configureServerWithArmeriaSettings;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.google.common.collect.ImmutableList;
+import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.metric.MeterIdPrefixFunction;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.spring.ArmeriaSettings;
+import com.linecorp.armeria.spring.InternalServices;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.ActiveProfiles;
+
+import javax.inject.Inject;
+
+@SpringBootTest(classes = ArmeriaConfigurationUtilTest.TestConfiguration.class)
+@ActiveProfiles({"local", "internalPortEqualToManagementPortTest"})
+@Timeout(10)
+class ArmeriaConfigurationUtilTest {
+
+    @SpringBootApplication
+    static class TestConfiguration {
+        @Bean
+        public Server armeriaServer() {
+            return Server.builder().service("/test", (ctx, req) -> HttpResponse.of(HttpStatus.OK)).build();
+        }
```

Comment:
I intentionally added a bean of type `Server` to prevent the automatic configuration of the [`armeriaServer`](https://github.com/line/armeria/blob/2bfe424b652a16f603e2ba1ca075bf992f80f24e/spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/spring/AbstractArmeriaAutoConfiguration.java#L67C19-L67C32) bean.